assets/js/gridtransition.js

```diff
@@ -1,36 +1,201 @@
     const productContainer = document.getElementById('productContainer');
-
     loadTranslations(getLanguage());
+    const urlParams = new URLSearchParams(window.location.search);
+    const currentPage = parseInt(urlParams.get('page')) || 1; // Lấy trang hiện tại từ URL
 
-    // Hàm hiển thị dạng Grid
-    function showGridView() {
-        fetch('../../src/layouts/grid.html')
-            .then(response => response.text())
-            .then(data => {
-                productContainer.innerHTML = data;
-                loadTranslations(getLanguage() || 'vi');
+    // Biến lưu trữ các bộ lọc
+    let selectedType = null;
+    let selectedPriceRange = null;
+    let selectedColor = null;
+
+    // Hàm lấy tất cả sản phẩm từ API
+    function fetchProducts() {
+        const language = localStorage.getItem('userLanguage');
+        const url = `http://localhost:3001/${language}_products`; // Không có phân trang ở đây, lấy toàn bộ sản phẩm
+        return fetch(url)
+            .then(response => {
+                if (!response.ok) {
+                    throw new Error("Network response was not ok");
+                }
+                return response.json(); // Trả về danh sách tất cả sản phẩm
             });
     }
 
-    // Hàm hiển thị dạng List
-    function showListView() {
-        fetch('../../src/layouts/list.html')
-            .then(response => response.text())
-            .then(data => {
-                productContainer.innerHTML = data;
-                loadTranslations(getLanguage() || 'vi');
-            });
+    // Hàm hiển thị dạng Grid với phân trang phía client
+    function showGridView(page = 1, limit = 9) {
+        fetchProducts().then(products => {
+            // Lọc sản phẩm theo loại, giá và màu sắc
+            let filteredProducts = applyFilters(products);
+
+            // Tính tổng số trang dựa trên số sản phẩm đã lọc
+            const totalPages = Math.ceil(filteredProducts.length / limit);
+
+            // Chỉ hiển thị sản phẩm của trang hiện tại
+            const paginatedProducts = filteredProducts.slice((page - 1) * limit, page * limit);
+
+            const gridContent = paginatedProducts.map(product => `
+                <div class="product-card group">
+                    <div class="relative">
+                        <a href="#">
+                            <img class="img-product" src="${product.image}" alt="${product.name}" />
+                        </a>
+                        <a href="#" class="img-product-hover group-hover:bg-opacity-50"></a>
+                        <div class="product-card-btn-wrapper">
+                            <div class="flex justify-center items-center w-full sm:flex-nowrap flex-wrap space-y-1">
+                                <button class="product-add-btn">
+                                    <i class="fas fa-shopping-cart me-[5px]"></i>
+                                    <span data-i18n="outstanding_products.add_to_cart"></span>
```

Comment:
E xử lý luôn logic button add to card nha. Phần này ở logic màn hình home Hiển có xử lý rồi, em check và áp dụng qua phần list products luôn nha